components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/state/OAuthRequestStateValidator.java

```diff
@@ -137,7 +138,8 @@ private void validateInputParameters(OAuthMessage oAuthMessage) throws InvalidRe
                     OAuth2ErrorCodes.INVALID_REQUEST, OAuth2ErrorCodes.OAuth2SubErrorCodes.INVALID_CLIENT);
         }
 
-        if (StringUtils.isBlank(oAuthMessage.getRequest().getParameter(REDIRECT_URI))) {
+        if (StringUtils.isBlank(oAuthMessage.getRequest().getParameter(REQUEST_URI)) &&
```

Comment:
The request object is already validate as part of/par endpoint. If the request object doest contain redirect_uri par call will end in error and the flow cannot go further.

Reply:
request_uri will endup creating a request object in the oauth component as well. so does the request object in the oauth component have a validation for this? Also according to the spec is the redirect_uri a mandatory param for the request object?

Reply:
the creation of request object from request_uri is delegated to Object Builder extension. Currently we are maintaining a builder for OB use cases in OB. If you require a validation or even not have one, you can add that in the IS specific extension implementation. Since you have that control, you can validate redirect_uri as per IS requirements too.

Reply:
Fixed!